thoth/package_extract/image.py

```diff
@@ -281,6 +288,77 @@ def _run_apt_cache_show(
     return result
 
 
+def _get_lib_dir_symbols(root_dir):
+    to_ret = set({})
+
+    # We grep for '0 A' here because all exported symbols are outputted by nm like:
+    # 00000000 A GLIBC_1.x or:
+    # 0000000000000000 A GLIBC_1.x
+    command = f"nm -D {root_dir}/*so* | grep '0 A'"
+
+    # Output of Popen is byte like, I decode it to a string to be able to use string functions
+    out = subprocess.Popen(command, stdout=subprocess.PIPE, shell=False).stdout.read().decode("utf-8")
+
+    lines = out.split('\n')
+    for line in lines:
+        columns = line.split(' ')
+        if len(columns) > 2:
+            to_ret.add(columns[2])
+    return to_ret
+
+
+def _root_lib_symbols(path: str):
+    to_ret = set()
+    if os.path.exists(os.path.join(path, "usr/lib64")):
+        to_ret |= _get_lib_dir_symbols(os.path.join(path, "usr/lib64"))
+        to_ret |= _get_lib_dir_symbols(os.path.join(path, "lib64"))
+    elif os.path.exists(os.path.join(path, "usr/lib32")):
+        to_ret |= _get_lib_dir_symbols(os.path.join(path, "usr/lib32"))
+        to_ret |= _get_lib_dir_symbols(os.path.join(path, "lib32"))
+    elif os.path.exists(os.path.join(path, "usr/lib")):
+        to_ret |= _get_lib_dir_symbols(os.path.join(path, "usr/lib"))
+        to_ret |= _get_lib_dir_symbols(os.path.join(path, "lib"))
+    else:
+        raise FileNotFoundError("No usr libraries were found")
+
+    return to_ret
+
+
+def _ld_config_symbols(path: str):
+    to_ret = set()
+    with cwd(os.path.join(path, "etc/")), open("ld.so.conf", "r") as f:
+        for line in f.readlines():
+            paths = glob.glob(line[:-1])
```

Comment:
Why `:-1`?

Reply:
It is grabbing the newline character, I should probably use .strip instead